source/common/api/posix/os_sys_calls_impl.cc

```diff
@@ -149,32 +150,46 @@ bool OsSysCallsImpl::supportsIpTransparent() const {
   // https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/net/ipv6/ipv6_sockglue.c?id=6c46862280c5f55eda7750391bc65cd7e08c7535
   //
   // So, almost recent linux kernel supports both IP_TRANSPARENT and IPV6_TRANSPARENT options.
+  // But there are also has ipv4 only or ipv6 only scenarios.
   //
   // And these socket options need CAP_NET_ADMIN capability to be applied.
   // The CAP_NET_ADMIN capability should be applied by root user before call this function.
-  static const bool is_supported = [] {
+
+  int fd = 0;
+  int val = 0;
+  bool result = false;
```

Comment:
i don't think we need these local variables here? we can just define necessary variables in the lambda directly.

Reply:
yeah,tweaked some lambda. Thanks.